docs/react-v9/contributing/rfcs/react-components/convergence/extended-design-tokens.md

```diff
@@ -0,0 +1,455 @@
+  ColorTokens &
+  ButtonTokens; // Addition
+```
+
+### Other options considered
+
+| Option                                                          | Pros                                                                                                                                     | Cons                                                                                                                                                                                                                                               |
+| --------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Option A: Add all tokens to current FluentProvider**          | • Simple<br>• Builds on existing API<br>• No additional deps for partners                                                                | • Theme typings will get large as we'd include all component tokens<br>• Tokens object will get larger but not massive since we'd only ship values for component groups and not component tokens (these are optional slots)                        |
+| Option B.1: Create separate semantic FluentProvider             | • Separates Fluent Semantic Tokens (FSTs) from FluentProvider and Theme.<br>• Optional layering so partners only pull it in when needed. | • Requires extra layer<br>• May not be as intuitive as just using the existing provider already in applications<br>• May not provide much value since the components still need to have some level of awareness of the group and component tokens. |
+| Option B.2: Create separate semantic FluentProvider and package | • Same pros as above but with added flexibility around packaging and versioning                                                          | See above                                                                                                                                                                                                                                          |
+
+## Update components to optionally consume expanded tokens
+
+Once exposed in the theme, the new tokens could be accessed in existing components simply by updating existing references to alias tokens with var(), passing the more precise token as the first argument and retaining original alias token as the second.
+
+For example, instead of:
+
+```javascript
+const useRootBaseClassName = makeStyles({ backgroundColor: 'var(--colorNeutralForeground1)' });
+```
+
+We would use this:
+
+```javascript
+const useRootBaseClassName = makeStyles({
+  backgroundColor: 'var(--buttonBackgroundColorRest, var(--colorNeutralForeground1))',
```

Comment:
It might be a good opportunity to explore hashing tokens as proposed before in one of RFCs: `--buttonBackgroundColorRest` will be `--fserwf` (and considering an amount of rules) => will have less bundle size impact (at price of harder debugging).

Reply:
Wouldn't the bundle cost of long variable names be gzipped out?

Reply:
Here's the attempt at hashing the control and semantic tokens: https://github.com/microsoft/fluentui/pull/32304

It brought down the change to +18.319 kB (gzipped)

Reply:
bit late to the party, correctly if i'm wrong, but the hashing approach would make it impossible to use this for web platform (web-components etc) ? 

looking at gzipped data is a mistake as that's only "speed of transfer from server to client", what matters is the non gzipped number as that's what's the client is taxing with.